app/models/spreadsheet_service.rb

```diff
@@ -0,0 +1,31 @@
+require 'google/apis/sheets_v4'
+
+class SpreadsheetService
+  def self.get_data(spreadsheet_id)
+    service = Google::Apis::SheetsV4::SheetsService.new
+    service.key = ENV['GOOGLE_SHEET_API_KEY']
+
+    range = SpreadsheetService.get_range(spreadsheet_id)
+    begin
+      response = service.get_spreadsheet_values(spreadsheet_id, range)
+    rescue Exception => e
+      raise ActionController::BadRequest.new(), e.message()
```

Comment:
I guess we need to add _some_ error handling regarding the Spreadsheet import - catch the exception, log it, show some error to the user asking to contact the site admin?